### PR TITLE
feat(wallet-dashboard): show tx details in unstaking flow

### LINKED ISSUE #4258: Add tx details state to the end of unstaking flow
Add tx details state to the end of unstaking flow, take wallet as reference
Depends on https://github.com/iotaledger/iota/issues/4245

### PR DESCRIPTION
**Pull request by cpl121**
# Description of change

Add tx details to unstaking flow
Depends on #4422 

## Links to any relevant issues

Closes #4258 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

### CODE REVIEW

**Comment by cpl121** [comment]
I have doubts about [this](https://github.com/iotaledger/iota/pull/4433/files#diff-b1e79598d83c78204112721bdac70b93cc98a7456d2e6eeeffcee07bf76e9333R73)
Were we blocking the unstake button while it was pending for some reason?
If the user unstakes before staking starts, he will not earn rewards, but I think it is a bad idea to leave the user with the funds blocked until staking starts

File: apps/wallet-dashboard/components/Dialogs/Staking/StakeDialog.tsx [hunk 1]
```diff
@@ -173,6 +174,29 @@ export function StakeDialog({
...
+            addNotification('Unstake transaction was not created', NotificationType.Error);
+            return;
+        }
+        signAndExecuteTransaction(
+            {
+                transaction: transaction,
+            },
+            {
+                onSuccess: (tx) => {
+                    onSuccess?.(tx.digest);
```
Comment (panteleymonchuk) [inline apps/wallet-dashboard/components/Dialogs/Staking/StakeDialog.tsx:188]: I see some of onSuccess callbacks uses `iotaClient.waitForTransaction` which is async.

Should we await it before move to the next TransactionDetails screen?

Comment (cpl121) [reply apps/wallet-dashboard/components/Dialogs/Staking/StakeDialog.tsx:188]: As I see, this component is called with `onSuccess` in `apps/wallet-dashboard/app/(protected)/staking/page.tsx` and `apps/wallet-dashboard/app/(protected)/vesting/page.tsx,` and in both places the `onSuccess` is not async
Where do you say it is async?

Comment (panteleymonchuk) [reply apps/wallet-dashboard/components/Dialogs/Staking/StakeDialog.tsx:188]: up to my understanding waitfortransaction async

image

Comment (cpl121) [reply apps/wallet-dashboard/components/Dialogs/Staking/StakeDialog.tsx:188]: nft output claim is not related with unstaking flow

**Review by panteleymonchuk — APPROVED** [review]
LGTM 🚀

File: apps/wallet-dashboard/components/Dialogs/Staking/StakeDialog.tsx (hunk 1 above)
Comment (panteleymonchuk) [reply apps/wallet-dashboard/components/Dialogs/Staking/StakeDialog.tsx:188]: It's just an example that waitForTransaction is async function.
It's using in the onSuccess callbacks. My concern here was if we need to await it before move to the next screen.
But we discussed that we don't 🫡

**Review by begonaalvarezd — APPROVED** [review]

File: apps/wallet-dashboard/components/Dialogs/Staking/views/UnstakeView.tsx [hunk 2]
```diff
@@ -67,7 +70,7 @@ export function UnstakeView({
         error: delegatedStakeDataError,
     } = delegatedStakeDataResult;
 
-    const delegationId = extendedStake?.status === 'Active' && extendedStake?.stakedIotaId;
+    const delegationId = extendedStake?.stakedIotaId;
```
Comment (begonaalvarezd) [inline apps/wallet-dashboard/components/Dialogs/Staking/views/UnstakeView.tsx:73]: nice catch 🚀

Comment (marc2332) [reply apps/wallet-dashboard/components/Dialogs/Staking/views/UnstakeView.tsx:73]: 🚀

File: apps/wallet-dashboard/components/Dialogs/Staking/views/UnstakeView.tsx [hunk 3]
```diff
@@ -26,22 +26,25 @@ import {
...
 import { useMemo } from 'react';
-import { useCurrentAccount, useSignAndExecuteTransaction } from '@iota/dapp-kit';
+import { useCurrentAccount } from '@iota/dapp-kit';
 import { Loader, Warning } from '@iota/ui-icons';
 import { useUnstakeTransaction } from '@/hooks';
 import { ValidatorStakingData } from '@/components';
 import { DialogLayout, DialogLayoutFooter, DialogLayoutBody } from '../../layout';
+import { Transaction } from '@iota/iota-sdk/transactions';
 
 interface UnstakeDialogProps {
```
Comment (marc2332) [inline apps/wallet-dashboard/components/Dialogs/Staking/views/UnstakeView.tsx]: Mind renaming this while you are it?
```suggestion
interface UnstakeViewProps {
```

**Review by marc2332 — CHANGES_REQUESTED** [review]

**Review by marc2332 — APPROVED** [review]

**Review by marc2332 — APPROVED** [review]

**Pull request merged**